nixos/modules/services/torrent/flood.nix

```
@@ -0,0 +1,115 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.flood;
+in
+{
+  meta.maintainers = with lib.maintainers; [ thiagokokada ];
+
+  options.services.flood = {
+    enable = lib.mkEnableOption "flood";
+    package = lib.mkPackageOption pkgs "flood" { };
+    openFirewall = lib.mkEnableOption "" // {
+      description = "Whether to automatically open the necessary ports in the firewall.";
+      default = true;
+    };
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = "User account used to run flood.";
+      default = "flood";
+    };
+    group = lib.mkOption {
+      type = lib.types.str;
+      description = "Group account used to run flood.";
+      default = "flood";
+    };
+    port = lib.mkOption {
+      type = lib.types.int;
+      description = "Port to bind webserver.";
+      default = 3000;
+    };
+    host = lib.mkOption {
+      type = lib.types.str;
+      description = "Host to bind webserver.";
+      default = "0.0.0.0";
+    };
+    runDir = lib.mkOption {
+      type = lib.types.path;
+      description = "Directory to store Flood configuration. Needs to be readable/writable by Flood's user/group.";
+      default = "/var/lib/flood";
+    };
+    extraArgs = lib.mkOption {
+      type = with lib.types; listOf str;
+      description = "Extra arguments passed to `flood`.";
+      default = [ ];
+      example = [ "--baseuri=/" ];
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users = {
+      users = lib.optionalAttrs (cfg.user == "flood") {
+        flood = {
+          group = cfg.group;
+          isSystemUser = true;
+        };
+      };
+      groups = lib.optionalAttrs (cfg.group == "flood") {
+        flood = { };
+      };
+    };
+
+    systemd.services.flood = {
+      description = "A web UI for rTorrent with a Node.js backend and React frontend.";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      unitConfig = {
+        Documentation = "https://github.com/jesec/flood/wiki";
+      };
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
```

```suggestion
    systemd.services.flood = {
      description = "A web UI for rTorrent with a Node.js backend and React frontend.";
      after = [ "network.target" ];
      wantedBy = [ "multi-user.target" ];
      unitConfig = {
        Documentation = "https://github.com/jesec/flood/wiki";
      };
      serviceConfig = {
        DynamicUser = true;
```

There should be no reason for a static user/group allocation.

---

And actually the project description is

>  A modern web UI for various torrent clients with a Node.js backend and React frontend.

---

> There should be no reason for a static user/group allocation.

One issue is that in some cases you want a static user/group allocation, for example to reuse a `rtorrent` user to access its permission. In this case you can't use `DynamicUser`.

So I would prefer to keep the service as currently is.

---

The proper solution for this is `SupplementaryGroup = [ "rtorrent" ]` on the unit IMO.

---

@mweinelt Did all the proposed changes. I think in the worst case I can override the service for my own needs. Can you re-review?